python/pyiceberg/table/__init__.py

```diff
@@ -183,14 +185,17 @@ class TableScan(Generic[S], ABC):
     def __init__(
         self,
         table: Table,
-        row_filter: Optional[BooleanExpression] = None,
+        row_filter: Optional[Union[str, BooleanExpression]] = None,
         selected_fields: Tuple[str] = ("*",),
         case_sensitive: bool = True,
         snapshot_id: Optional[int] = None,
         options: Properties = EMPTY_DICT,
     ):
         self.table = table
-        self.row_filter = row_filter or AlwaysTrue()
+        if row_filter is None:
+            self.row_filter = AlwaysTrue()
+        else:
+            self.row_filter = parser.parse(row_filter)
```

Comment:
Nit: Maybe we could use a ternary operator here? 

```
self.row_filter = parser.parse(row_filter) if row_filter else AlwaysTrue()
```

Reply:
Good suggestion, I've split it into a separate method for now, that also takes care of the parsing.